sympy/physics/vector/tests/test_point.py

```diff
@@ -126,3 +128,67 @@ def test_point_partial_velocity():
     assert p.partial_velocity(N, u1) == A.x
     assert p.partial_velocity(N, u1, u2) == (A.x, N.y)
     raises(ValueError, lambda: p.partial_velocity(A, u1))
+
+def test_point_vel():
+    q1, q2 = dynamicsymbols('q1 q2')
+    N = ReferenceFrame('N')
+    B = ReferenceFrame('B')
+    S = ReferenceFrame('S')
+    O = Point('O')
+    Q = Point('Q')
+    Q.set_pos(O, q1 * N.x)
+    raises(ValueError , lambda : Q.vel(N)) #Velocity of Q is not defined
+    O.set_vel(N, q2 * N.y)
+    assert Q.vel(N) == Derivative(q1) * N.x + q2 * N.y  # Velocity of Q using O
+    P1 = Point('P1')
+    P1.set_pos(O, q1 * B.x)
+    raises(ValueError, lambda : P1.vel(B)) # O's velocity is defined in different frame
+    raises(ValueError, lambda : P1.vel(N)) # Pos vector is defined in different frame
+    P2 = Point('P2')
+    P2.set_pos(P1, q2 * B.z)
+    raises(ValueError, lambda : P2.vel(B)) # O's velocity is defined in different frame, and no
+    #point in between has its velocity defined
+    P3 = Point('P3')
+    P3.set_pos(P2, q1 * B.x)
+    raises(ValueError, lambda : P3.vel(B)) # O's velocity is defined in different frame, and no
+    #point in between has its velocity defined
+    P1.set_vel(B, 10 * q1 * B.x) #Defined P1's velocity
+    assert P3.vel(B) == (10 * q1 + Derivative(q1)) * B.x + Derivative(q2) * B.z # Tree traversal upto P1
```

Comment:
It is best to use:

```python
from sympy.physics.mechanics import dynamicsymbols
t = dynamicsymbols._t
q = dynamicsymbols('q')
q.diff(t)
```
to differentiate dynamics symbols.

Reply:
Okay , i'll change that

Reply:
Is this change really needed? As Derivative also does the same thing to dynamicsymbols i.e diff wrt t

Reply:
Yes.